src/fitnesse/slim/fixtureInteraction/DefaultInteraction.java

```diff
@@ -3,88 +3,144 @@
+		Object[] initargs = ConverterSupport.convertArgs(args,
+				constructor.getParameterTypes());
+
+		return newInstance(constructor, initargs);
+	}
+
+
+	protected Class<?> searchPathsForClass(List<String> paths, String className) {
+		Class<?> k = getClass(className);
+
+		if (k != null) {
+			return k;
+		}
+		for (String path : paths) {
+			k = getClass(path + "." + className);
+			if (k != null) {
+				return k;
+			}
+		}
+		throw new SlimError(String.format("message:<<%s %s>>", SlimServer.NO_CLASS, className));
+	}
+
+	protected Class<?> getClass(String className) {
+		try {
+			return Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			return null;
+		}
+	}
+
+	protected Constructor<?> getConstructor(Class<?> clazz,
+			Object[] args) {
+
+		for (Constructor<?> constructor : clazz.getConstructors()) {
+			Class<?>[] arguments = constructor.getParameterTypes();
+			if (arguments.length == args.length) {
+				return constructor;
+			}
+		}
+		return null;
+	}
+
+	protected Object newInstance(Constructor<?> constructor, Object... initargs) throws InvocationTargetException, InstantiationException, IllegalAccessException {
+		return constructor.newInstance(initargs);
+	}
+
+	@Override
+	public MethodExecutionResult findAndInvoke(String methodName, Object instance, Object... args) throws Throwable {
+		Method method = findMatchingMethod(methodName, instance.getClass(), args.length);
+		if (method != null) {
+			return this.invokeMethod(instance, method, args);
+		}
+		return MethodExecutionResult.noMethod(methodName, instance.getClass(), args.length);
+	}
+	
+	protected Method findMatchingMethod(String methodName, Class<?> k, int nArgs) {
+		Method[] methods = k.getMethods();
+		if(methods == null) {
+			methods = k.getMethods();
+		}
+
+		for (Method method : methods) {
+			boolean hasMatchingName = method.getName().equals(methodName);
+			boolean hasMatchingArguments = method.getParameterTypes().length == nArgs;
+			if (hasMatchingName && hasMatchingArguments) {
+				return method;
+			}
+		}
+		return null;
+	}
+
+	protected MethodExecutionResult invokeMethod(Object instance, Method method, Object[] args) throws Throwable {
+		Object[] convertedArgs = convertArgs(method, args);
+		Object retval = callMethod(instance, method, convertedArgs);
+		Class<?> retType = method.getReturnType();
+		return new MethodExecutionResult(retval, retType);
+	}
+
+	protected Object[] convertArgs(Method method, Object[] args) {
+		Type[] argumentParameterTypes = method.getGenericParameterTypes();
+		return ConverterSupport.convertArgs(args, argumentParameterTypes);
+	}
+
+	protected Object callMethod(Object instance, Method method, Object[] convertedArgs) throws Throwable {
+		try {
+			Object result;
+			if (instance instanceof InteractionAwareFixture) {
+				// invoke via interaction, so it can also do its thing on the aroundMethod invocation
+				result = ((InteractionAwareFixture)instance).aroundSlimInvoke(this, method, convertedArgs);
```

Comment:
Previously this code explicitly called the method using the interaction (and not directly as a method call, and that was what the comment refers to, so now it no longer makes any sense: aroundSlimInvoke is NOT called via the interaction, but normally/directly). 
Previously the code making this call was not part of the interaction, so there was a bit more delegation going on. The code making the call could not rely on what the interaction would do exactly. Now the call is done from inside the interaction, so that uncertainty no longer exists. So maybe just removing the comment is a fine solution.

But now I wonder whether there will be an infinite loop with `InteractionAwareFixture.aroundSlimInvoke()`. Previously it would not do the actual method call itself, but delegate that to the interaction again (that's why it's passed as first argument, and that's also the recommended approach in `InteractionAwareFixture.aroundSlimInvoke`'s documentation). Now in the new setup the interaction calls `aroundSlimInvoke` so the fixture calling the interaction seems like a recipe for an infinite loop.

I'll dig a bit deeper what the current status is and what can be done about this.